RuntimeException (FileNotFound) in kotlin-client (swagger 3.0.0-SNAPSHOT-79)

I'm getting the following error when trying to generate **kotlin-client** with version `swagger-codegen-cli-3.0.0-20180615.112032-79.jar`

```
Exception in thread "Thread-1" java.lang.RuntimeException: Could not generate supporting file 'Supporting
        templateFile: README.mustache
        folder:
        destinationFilename: README.md
'
        at io.swagger.codegen.DefaultGenerator.generateSupportingFiles(DefaultGenerator.java:614)
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:736)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:331)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.io.FileNotFoundException: /v2/kotlin-client/v2/kotlin-client/README.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:357)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:343)
        at io.swagger.codegen.DefaultGenerator.getHandlebars(DefaultGenerator.java:1027)
        at io.swagger.codegen.DefaultGenerator.generateSupportingFiles(DefaultGenerator.java:581)
        ... 3 more
```

--- a/modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultGenerator.java
+++ b/modules/swagger-codegen/src/main/java/io/swagger/codegen/DefaultGenerator.java
@@ -575,17 +575,12 @@ public class DefaultGenerator extends AbstractGenerator implements Generator {
                 }
 
                 if(ignoreProcessor.allowsFile(new File(outputFilename))) {
-                    templateFile = templateFile.replaceAll("//", "/").replace('/', File.separatorChar);
                     if (templateFile.endsWith("mustache")) {
-                        String templateName = templateFile;
-                        final com.github.jknack.handlebars.Template hTemplate = getHandlebars(templateName.replace(config.templateDir(), StringUtils.EMPTY));
+                        final com.github.jknack.handlebars.Template hTemplate = getHandlebars(templateFile);
                         String rendered = hTemplate.apply(bundle);
                         writeToFile(outputFilename, rendered);
-
-                        // writeToFile(outputFilename, tmpl.execute(bundle));
                         files.add(new File(outputFilename));
                     } else {
-                        templateFile = templateFile.replace("\\", "/");
                         InputStream in = null;
 
                         try {
@@ -1011,19 +1006,19 @@ public class DefaultGenerator extends AbstractGenerator implements Generator {
     }
 
     private com.github.jknack.handlebars.Template getHandlebars(String templateFile) throws IOException {
-        if (templateFile.startsWith(config.templateDir())) {
-            templateFile = StringUtils.replaceOnce(templateFile, config.templateDir(), StringUtils.EMPTY);
+        templateFile = templateFile.replace(".mustache", StringUtils.EMPTY).replace("\\", "/");
+        String templateDir = config.templateDir().replace(".mustache", StringUtils.EMPTY).replace("\\", "/");
+        if (templateFile.startsWith(templateDir)) {
+            templateFile = StringUtils.replaceOnce(templateFile, templateDir, StringUtils.EMPTY);
         }
         TemplateLoader templateLoader = null;
         if (config.additionalProperties().get(CodegenConstants.TEMPLATE_DIR) != null) {
-            templateLoader = new FileTemplateLoader(config.templateDir(), ".mustache");
+            templateLoader = new FileTemplateLoader(templateDir, ".mustache");
         } else {
-            templateLoader = new ClassPathTemplateLoader("/" + config.templateDir().replace("\\", "/"), ".mustache");
+            templateLoader = new ClassPathTemplateLoader("/" + templateDir, ".mustache");
         }
         final Handlebars handlebars = new Handlebars(templateLoader);
         config.addHandlebarHelpers(handlebars);
-
-        templateFile = templateFile.replace(".mustache", StringUtils.EMPTY).replace("\\", "/");
         return handlebars.compile(templateFile);
     }
 

